### ISSUE TITLE
The future of cryptii

### ISSUE THREAD

**Issue by ffraenz** [cryptii/cryptii#129]
**TL;DR**: We are working on a complete rewrite of the app. If your issue got closed with a reference to this one, it will be considered in the development process.

---

cryptii.com is an educational and productive web app offering modular conversion, encoding and encryption methods. It received its last major overhaul in 2017 and is now celebrating its 10th birthday on the web. Today the web app gets requested over 10k times a day.

We've learned so much during that time and we'd like to put it into good use. That's why we are already working on a complete redesign and rewrite of the application behind the scenes.

### Why a complete rewrite?

One of the main reasons for rebuilding the app from the ground up is the separation of the app's source from the source of the translation methods it offers. Until now the entire app is powered by a single script file including all translation methods wether they are in use or not. Because of this the script file size was an important factor when deciding on wether to include a certain translation method or not. For example we were not able to merge an amazing pull request introducing a database of 195 KB of different spelling alphabets because of the file size increase it would cause (see #82).

In the next major release we will dynamically load the source code of a method only when it is required by the current pipeline configuration. On top of that, translation methods will be developed and bundled in repositories seperate from the main app's source code. This will allow you in the future to develop and publish your own translation methods without issuing pull requests on the main repository.

Apart from that we have much more to share with you in the future.

### How to get involved?

If you want to get involved in the next cryptii, fill out this short survey and sign up to the user testing pool, developer pool and/or pool of writers on the last page: https://forms.gle/wU11nYMy73EiDFk76. We really appreciate it.

### What about this repository?

Concerning the current v4 of the app, we'd like to update it and its build tools, work up the open issues and pull requests and publish a final release v4.0.10 for it. During this process a lot of issues will be closed with a reference to this one. If that's the case for you, your contribution will not be lost. We will collect the ideas and suggestions you've made and consider them during the development of the next cryptii.

---

We'll keep this thread updated. If you have any questions, post them below or reach out to us via hello@cryptii.com.

Thank you for your support!

**Comment by ffraenz** [comment]
Update: We have published the first alpha preview of the cryptii successor ciphereditor. Head over to [ciphereditor.com](https://ciphereditor.com) and the [ciphereditor repository on GitHub](https://github.com/wierkstudio/ciphereditor) and give it a try. We are listening for feedback on our [community Discord server](https://discord.gg/v5wNewjdxm).

**Comment by q2dg** [comment]
Well...last Ciphereditor commit was from two years ago...it seems its development has been stalled a bit, hasn't it?

**Comment by ffraenz** [comment]
You’re right—the project has been stalled. It’s not for lack of interest but a shift in priorities: I couldn’t see a clear path to monetization for Ciphereditor, so I had to focus on projects where that isn’t an unknown.

There are also some bigger architectural decisions I want to settle first (e.g., how the graph computes, potentially via a Petri-net, style model—and how extensions should be structured, since the current package format is outdated).

The sites aren’t dead though. cryptii.com still gets many daily visitors and there’s an active Discord community.
